iothub/device/src/ModuleClient.cs

```diff
@@ -845,30 +860,20 @@ private async Task<MethodResponse> InvokeMethodAsync(Uri uri, MethodRequest meth
                         SslProtocols = TlsVersions.Instance.Preferred,
                     };
 #else
-                    httpClientHandler = new WebRequestHandler();
+#pragma warning disable CA2000 // Dispose objects before losing scope
+                    // This object is given to an HTTP client to use, so it can't be disposed yet
+                    var httpMessageHandler = new WebRequestHandler();
+#pragma warning restore CA2000 // Dispose objects before losing scope
                     ((WebRequestHandler)httpClientHandler).ServerCertificateValidationCallback = (sender, certificate, chain, errors) =>
                     {
                         return customCertificateValidation(sender, certificate, chain, errors);
                     };
-#endif
-                }
-
-                var context = new PipelineContext()
-                {
-                    ProductInfo = new ProductInfo
-                    {
-                        Extra = InternalClient.ProductInfo
-                    }
-                };
 
-                var transportSettings = new Http1TransportSettings();
-                //We need to add the certificate to the httpTransport if DeviceAuthenticationWithX509Certificate
-                if (InternalClient.Certificate != null)
-                {
-                    transportSettings.ClientCertificate = InternalClient.Certificate;
+                    transportSettings.HttpClient = new HttpClient(httpMessageHandler);
```

Comment:
Isn't this going to use the client passed in through transport settings?

Reply:
Good catch

Reply:
Actually, this is a bit of an odd case. We already don't check if the user passed in an HttpTransportSettings class here because modules only use MQTT/AMQP except for this operation. We could create a ModuleDirectMethodTransportSettings class like we do with FileUploadTransportSettings, but that strikes me as more work than necessary for v1 since it will soon be replaced. 

I'm inclined to just not use the custom HTTP client for this particular case to keep v1 simple. Thoughts?

Reply:
So, this case essentially doesn't even expect a transport setting since the client is initialized over either MQTT or AMQP. Since we already have the precedent set for not looking at transport settings for this case, I agree with your approach.

On a related note - I thought we did allow module clients over HTTP, but I guess if we do it's only IoT device modules and not IoT Edge modules (which uses this method). Is that right?

Reply:
The v2 module client is also set up similarly to ignore http settings. We didn't have a comment for that, so I added it in: https://github.com/Azure/azure-iot-sdk-csharp/pull/3297

We should update the FileUploadTransportSettings to be more generic on our v2 client and have both device and module clients reference it for their HTTP operations.

Reply:
PS: I don't plan to merge #3297 in. I created that to highlight the behavior as you work on your changes.

Reply:
I tried removing that Java SDK HTTP check and running a telemetry test, but the hub threw a 404 error, so it looks like IoT hub doesn't support HTTP module operations